packages/block-library/src/image/block.json

```diff
@@ -72,6 +72,11 @@
 	},
 	"supports": {
 		"anchor": true,
+		"color": {
+			"duotone": "img",
```

Comment:
I don't like that the duotone can be a selector like that. It feels like something that should be avoided for now (v1) or  thought in a more generic way. For instance we've been considering whether the value of each support can be generated as a CSS variable to be applied by the block style itself.

Reply:
Can you give me an example of what a "good" selector would look like? Where are these CSS variables being generated? Can you give me an example of a CSS variable that might be generated too?

Reply:
The CSS variables PR is being worked on here https://github.com/WordPress/gutenberg/pull/29714
The idea is that instead of generating the whole style, the block supports just generates CSS variable assignments at the root level of the block and the CSS file of the block contains styles to apply that variable at the right place.

Reply:
How does creating CSS variables relate to the selector not being good? You still have to select where the styles are applied. Can you give an example of what you are talking about?

Reply:
I'll explain with the "colors" example as it's simpler. Imagine you use the block supports `color` in a block but you want the text color to apply to a specific element of the block and not the whole block. So do add this bit of CSS to your block's style file.

```
.my-block .my-element {
   color: var( --wp-style-color-text ); 
}
```

and the block supports adds the following HTML attribute to the block wrapper

```
style="--wp-style-color-text: red"
```

--- 
Obviously it's harder for duotone styles, but I was wondering if we can come up with a way to apply the effect as one or multiple CSS variables.

Reply:
Ah, so the selector happens in the CSS, not the block.json. Thanks for the example—what you were saying makes more sense now. I'll see what I can come up with.

Reply:
So to circle back here, I think we should avoid the CSS variables approach for now (I explained on the linked PR) which means we're back to:

 1- only support duotone for blocks where it applies to the wrapper directly
 2- Have a selector defined somehow like proposed in the PR

I'm not against 2 personally but if we can have this PR be valuable in one block and ship it without 2. That would give us time to think about that API better. The reason I'm saying that is because it's a precedent for a block support and I would like us to do it right.

Reply:
1 isn't going to work for any blocks right now. Even the image block is wrapped in a container, so the duotone would get applied to the caption too which we don't want. Maybe we might just have to do an `__experimental` prefix on the block supports or something?

Reply:
> Unfortunately 1 isn't going to work for any blocks right now. Even the image block is wrapped in a container, so the duotone would get applied to the caption too which we don't want. Maybe we might just have to do an __experimental prefix on the block supports or something? 

This works for me yes.

also cc @nosolosw @ockham  about how to define "selectors" for block supports, I think this is something you may have been thinking about as well separately?

Reply:
Took a quick look at the PR. There's a lot of fun going around here! :clap: 

Yes, one of the tasks at https://github.com/WordPress/gutenberg/issues/28913 is to look into how different block supports can target different inner elements of the block. That's a task no one has started to look at (that I know of).

The string may work for me. The only reason I'm not fully convinced is that, at some point, we talked about other things block supports may need:

- Provide defaults via the block supports declaration instead of having to write the `defaults.style.duotone: {..}` to provide defaults. Not sure if there's enough interest for this.
- Provide info about where the UI controls should be hiden/shown: `color.text: "block-sidebar"`. It seems this is no longer needed.

so it seems like we may want to have room for grow in the future. An alternate approach could be:

```json
"supports": {
  "color": {
    "duotone": {
      "selector": "this would either be inner element"
    }
  },
  "__experimentalSelector": "the wrapper selector ― not necessary if the block uses a regular class"
}
```

Then, there's also another thing to consider. In addition to what we already have for `__experimentalSelector`, we're going to need to have something for the elements being introduced at https://github.com/WordPress/gutenberg/issues/29891 See example for how we could do that at https://github.com/WordPress/gutenberg/pull/30541/files#diff-801846106160a8e52c94f9316006260c2e6438ba5e6ee086b93d9f5f074122bcR32

This is a long way to say that perhaps it's in our best interest to do `__experimentalDuotone` with the string at the moment and revisit once those things are clearer (or viceversa if the other things land first).